### PR TITLE
Hide shift 'edit' and 'delete' options if shift taken

### PR DESCRIPTION
**Pull request by ncbui**
## What type of PR is this? (check all applicable)

- [] Feature
- [] Refactor
- [] Bug Fix
- [x] Optimization
- [] Documentation Update
- [] Other (describe: )

## Description of what PR does
This completes step 2 of #45 . This removes the ability to edit shifts that have been taken. In the org view, orgs cannot see an edit link to the shift. If user tries to edit a shift via url, the shift controller will redirect to shifts page with error message 

## Related PRs and/or Issues (if any)
- Fixes step 2 or #45 
-

## QA Instructions, Screenshots, Recordings

Navigate to 'http://127.0.0.1:3000/organizations/:id' to see how it looks on org page
Or try to directly access an edit page for a filled shift 'http://127.0.0.1:3000/shifts/:id/edit'

Screen Shot 2021-04-28 at 5 25 03 PM
Screen Shot 2021-04-28 at 5 24 48 PM

## Added tests?

- [ ] yes
- [ ] no, because they aren't needed _(please include reasons for why tests aren't needed)_
- [ ] no, because I need help
- [ x] no, they will be added later (please create an Issue for it)

## Added to documentation?

- [ ] Yes, project README
- [ x] No documentation needed

### CODE REVIEW

**Review by Timilehin08 — APPROVED** [review]
Great Work! @ncbui

**Comment by saradotlog** [comment]
It works well @ncbui 
Just a thought... Wouldn't it be nicer if the edit button is also hidden in the shifts page if the shift is filled?

**Comment by ncbui** [comment]
Thanks for the suggestion @saradotlog , I'll fix it on the shifts page

**Comment by ncbui** [comment]
Have included the shifts page, please re-review when you have the chance @saradotlog

File: app/views/organizations/show.html.erb [hunk 1]
```diff
@@ -33,7 +33,7 @@
                         <div class="col-7"><% unless shift.shift_open %>Worker Name<% end %></div>
                     </div>
                     <div class="text-right">
-                        <%= link_to "Edit", edit_shift_path(shift.id), class: "btn" %>
+                        <% if shift.shift_open %><%= link_to "Edit", edit_shift_path(shift.id), class: "btn" %> <% end %>
                         <%= link_to "Delete", shift_path(shift.id), method: :delete, data: { confirm: "Are you sure?" }, class: "btn" %>
```
Comment (msarit) [inline app/views/organizations/show.html.erb:37]: I think you should wrap the `<% if %>` block around the `Delete` button as well.

File: app/controllers/shifts_controller.rb [hunk 2]
```diff
@@ -25,6 +25,10 @@ def new
   end
 
   def edit
+    unless @shift.shift_open?
```
Comment (msarit) [inline app/controllers/shifts_controller.rb:28]: Add a similar guard to the actual `shift#update` action in this controller as well 🙏🏾 
What you added here just covers the Edit Form, but not the update action.

**Review by msarit — CHANGES_REQUESTED** [review]

**Comment by msarit** [comment]
Also resolve conflicts, @ncbui 😅

File: app/views/organizations/show.html.erb (hunk 1 above)
Comment (ncbui) [reply app/views/organizations/show.html.erb:37]: Thanks @msarit , will update

File: app/controllers/shifts_controller.rb (hunk 2 above)
Comment (ncbui) [reply app/controllers/shifts_controller.rb:28]: Thanks for the catch @msarit, have made this a before_action that covers edit, update, and destroy actions

**Comment by saradotlog** [comment]
hey @ncbui! Nice! I can't see the edit/delete buttons on the shifts page, but I can still see the delete one in the organization/id page, I guess it makes sense orgs cannot delete the shift from any page?

**Comment by GabrielleEaston** [comment]
@ncbui I just noticed if I go to http://localhost:3000/organizations/1, I have an option to delete the filled shift!
If I am at http://localhost:3000/shifts, everything works as intended.

**Comment by ncbui** [comment]
Thanks for catching that @saradotlog , have updated

**Review by msarit — APPROVED** [review]

**Pull request merged**